x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshots.java

```diff
@@ -234,6 +236,18 @@
         Setting.Property.NotCopyableOnResize
     );
 
+    /**
+     * Index setting used to indicate if the snapshot that is mounted as an index should be deleted when the index is deleted. This setting
+     * is only set for indices mounted in clusters on or after 8.0.0. Once set this setting cannot be updated.
+     */
+    public static final Setting<Boolean> DELETE_SEARCHABLE_SNAPSHOT_ON_INDEX_DELETION = Setting.boolSetting(
+        SEARCHABLE_SNAPSHOTS_DELETE_SNAPSHOT_ON_INDEX_DELETION,
+        false,
+        Setting.Property.IndexScope,
```

Comment:
Should we add "Final" too, in case the searchable snapshot is closed? Mostly a nit because it is private anyway.

Reply:
I added `Final` too but I also implemented a check that prevents to change the value of this setting when restoring a searchable snapshots index (as Final does not prevent changes at restore time)